src/flip.jl

```diff
@@ -133,8 +134,9 @@ function flip_chain(graph::BaseGraph,
             # go back to the previous partition
             partition = partition.parent
         end
-        scores = get_scores(graph, partition, score_keys, steps_taken, proposal)
-        push!(all_scores, scores)
+        score_vals = score_partition_from_proposal(graph, partition, proposal, scores)
+        push!(chain_scores, score_vals)
         steps_taken += 1
     end
+    return chain_scores

```

Comment:
the _chain_scores_ array is going to be of length _num_steps_ + 1, right? (because we take _num_steps_ steps and the original partition at the front. How do we feel about that?

The Python implementation gets around this by calling the original partition the first step, so if you run a chain of num_steps=1 you haven't actually taken a step but the _chain_scores_ equivalent would be of length num_steps=1 atleast.

I prefer this though, although the _chain_scores_ arr being of length _num_steps_ + 1 felt awkward at first I think this is better because you are actually taking _num_steps_ after all. Might be something to think about when calling the chain from Python though because it has the different convention.

Reply:
That's a good point! I think that you're right - 1 step should be considered as 1 "move" between states, rather than the total number of states traversed by the chain.

Reply:
I'm going to edit `get_scores_at_step` so passing in `step=0` returns the scores at the initial partition.

Reply:
thats a great idea!